src/util/arrayView.js

```diff
@@ -0,0 +1,74 @@
+// @flow
+
+import {type PluckFn} from "./sortBy";
+import sortBy from "./sortBy";
+
+export class ArrayView<T> {
+  _array: Array<T>;
+  _filteredArray: Array<T>;
+  _rowsPerPage: number;
+  _pages: number;
+
+  constructor(data: $ReadOnlyArray<T>) {
+    this._array = JSON.parse(JSON.stringify(data)); //Deep Copy
+    this._filteredArray = this._array;
+    this._rowsPerPage = this._array.length;
+    this._pages = 1;
+  }
+
+  /**
+    pluckArg: a function of an object that returns the element by which to sort
+    order: "ascending" or "descending"
+   */
+  sortBy(pluckArg: PluckFn<T>, order: "ascending" | "descending") {
+    this._array = sortBy(this._array, pluckArg);
+    this._filteredArray = sortBy(this._filteredArray, pluckArg);
+    if (order === "descending") {
+      this._array.reverse();
+      this._filteredArray.reverse();
+    }
+  }
+
+  /**
+    Removes any existing filters and applies the new filter if provided.
+
+    filterFn: a function that returns true for objects that should be included, omit to remove filter
+    return: the new number of pages
+   */
+  filter(filterFn?: (T) => boolean): number {
+    this._filteredArray = filterFn ? this._array.filter(filterFn) : this._array;
+    this._pages = Math.ceil(this._filteredArray.length / this._rowsPerPage);
+    return this._pages;
+  }
+
+  /**
+    rowsPerPage: how many rows should be displayed per page, +Infinity to display all rows on one page
+    return: the new number of pages
+   */
+  setRowsPerPage(rowsPerPage: number): number {
+    rowsPerPage = Math.min(rowsPerPage, this._filteredArray.length);
+    rowsPerPage = Math.max(rowsPerPage, 1);
+    rowsPerPage = parseInt(rowsPerPage.toFixed(0));
```

Comment:
This line is to remove decimals since JS doesn't have an int type, because lol I forgot about Math.round
decimals would cause not good things to happen later on if we stored them, so this is just for robustness.

Reply:
`parseInt(1.2)` already returns `1` tho

Reply:
today I learned